include/k4arecord/playback.hpp

```diff
@@ -12,13 +12,115 @@
 
 #include <algorithm>
 #include <chrono>
+#include <memory>
 #include <string>
 #include <vector>
 
 namespace k4a
 {
 
-/** \class playback playback.hpp
+/** \class data_block playback.hpp <k4arecord/playback.hpp>
+ * Wrapper for \ref k4a_playback_data_block_t
+ *
+ * \sa k4a_playback_data_block_t
+ */
+class data_block
+{
+public:
+    /** Creates a data_block from a k4a_playback_data_block_t
+     * Takes ownership of the handle, you should not call k4a_playback_data_block_release on the handle after
+     * giving it to the data_block; the data_block will take care of that.
+     */
+    data_block(k4a_playback_data_block_t handle = nullptr) noexcept : m_handle(handle) {}
+
+    // No Copies allowed
+    data_block(const data_block &) = delete;
+    data_block &operator=(const data_block &) = delete;
+
+    /** Moves another data_block into a new data_block
+     */
+    data_block(data_block &&other) noexcept : m_handle(other.m_handle)
+    {
+        reset();
+        m_handle = other.m_handle;
+        other.m_handle = nullptr;
+    }
+
+    ~data_block()
+    {
+        reset();
+    }
+
+    /** Moves another data_block into this data_block; other is set to invalid
+     */
+    data_block &operator=(data_block &&other) noexcept
+    {
+        reset();
+        if (this != &other)
+        {
+            m_handle = other.m_handle;
+            other.m_handle = nullptr;
+        }
+        return *this;
+    }
+
+    /** Invalidates this data_block
+     */
+    data_block &operator=(std::nullptr_t) noexcept
+    {
+        return *this;
+    }
```

Comment:
This doesn't seem to do anything - I don't think we want this to have pointer semantics at all, so I'd just delete this function entirely